sensor_msgs/include/sensor_msgs/point_cloud_conversion.hpp

```diff
@@ -36,12 +36,23 @@
 #ifndef SENSOR_MSGS__POINT_CLOUD_CONVERSION_HPP_
 #define SENSOR_MSGS__POINT_CLOUD_CONVERSION_HPP_
 
+
 #include <sensor_msgs/msg/point_cloud.hpp>
 #include <sensor_msgs/msg/point_cloud2.hpp>
 #include <sensor_msgs/point_field_conversion.hpp>
 
 #include <string>
 
+#define POINT_CLOUD_DEPRECATION_MESSAGE \
+  "PointCloud is deprecated in Foxy for PointCloud2. This whole header will be removed."
+#ifdef _MSC_VER
+#pragma message(POINT_CLOUD_DEPRECATION_MESSAGE)
+#else
+#ifndef RCUTILS_SKIP_WARNING
```

Comment:
This is specifically to the bug in g++ that the preprocessor doesn't respect the pragmas. So it's better to have the pragma suppress the warning on Windows instead of compiling it out.

 https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53431